### PR TITLE
Replace symlink resolution with realpath

### PR DESCRIPTION
**Pull request by arkalyanms**
Credit: @jasonmalinowski 

Fixes https://github.com/dotnet/vscode-csharp/issues/6218

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,9 @@
 - Debug from .csproj and .sln [#5876](https://github.com/dotnet/vscode-csharp/issues/5876)
 
 ## Latest
+* Fix URIFormatException during dotnet SDK resolution when symlinks are involved (PR: [#6230](https://github.com/dotnet/vscode-csharp/pull/6230))
```
Comment (jasonmalinowski) [inline CHANGELOG.md:12]:
```suggestion
* Fix UriFormatException during dotnet SDK resolution when symlinks are involved (PR: [#6230](https://github.com/dotnet/vscode-csharp/pull/6230))
```

**Review by jasonmalinowski — APPROVED** [review]
:shipit:, assuming it actually works.

File: CHANGELOG.md (hunk 1 above)
Comment (jasonmalinowski) [reply CHANGELOG.md:12]: (feel free to ignore this if you have green CI)

**Comment by Cosifne** [comment]
Tested on Ubuntu 22.04
No Uri problem
image

**Review by Cosifne — APPROVED** [review]

**Pull request merged**